generator/insn_vdvs2fs1vm.go

```diff
@@ -0,0 +1,61 @@
+package generator
+
+import (
+	"fmt"
+	"math"
+	"strings"
+)
+
+func (i *insn) genCodeVdVs2Fs1Vm() []string {
+	vdWidening := strings.HasPrefix(i.Name, "vfw")
+	vs2Widening := strings.HasSuffix(i.Name, "wf")
+	vdSize := iff(vdWidening, 2, 1)
+
+	sews := iff(vdWidening || vs2Widening, floatSEWs[:len(floatSEWs)-1], floatSEWs)
+	combinations := i.combinations(
+		iff(vdWidening || vs2Widening, wideningMULs, allLMULs),
+		sews,
+		[]bool{false, true},
+	)
+	res := make([]string, 0, len(combinations))
+
+	for _, c := range combinations {
+		builder := strings.Builder{}
+		builder.WriteString(i.gTestDataAddr())
+		builder.WriteString(i.gWriteRandomData(LMUL(1)))
+		builder.WriteString(i.gLoadDataIntoRegisterGroup(0, LMUL(1), SEW(8)))
+
+		builder.WriteString(c.comment())
+
+		vdEMUL1 := LMUL(math.Max(float64(int(c.LMUL)*vdSize), 1))
+		vdEEW := c.SEW * SEW(vdSize)
+		vd := int(vdEMUL1)
+		vs2 := vd * 2
+		builder.WriteString(i.gWriteRandomData(vdEMUL1))
+		builder.WriteString(i.gLoadDataIntoRegisterGroup(vd, vdEMUL1, SEW(8)))
+
+		builder.WriteString(i.gWriteTestData(true, c.LMUL1, c.SEW, 0))
+		builder.WriteString(i.gLoadDataIntoRegisterGroup(vs2, c.LMUL1, c.SEW))
+
+		cases := i.testCases(true, c.SEW)
+		for a := 0; a < len(cases); a++ {
+			builder.WriteString("# -------------- TEST BEGIN --------------\n")
+			switch c.SEW {
+			case 32:
+				builder.WriteString(fmt.Sprintf("li s0, 0x%x\nfmv.w.x f0, s0\n", convNum[uint32](cases[a][1])))
```

Comment:
Vs2 test data should be fetched from `cases[x][1]`, and fs1 should be fetched from `cases[x][0]`, not opposed.

Reply:
Also, you may want to split this line into two lines, for readability.